src/features/deployProcess/slices/stepperSlice.tsx

```
@@ -70,32 +104,42 @@ export const fetchProductStatus = createAsyncThunk<Product, string>(
       );
       return response.data as Product;
     } catch (err: any) {
-      return thunkAPI.rejectWithValue(
-        err.response?.data?.message || err.message
+      return thunkAPI.rejectWithValue(err.response?.data?.message || err.message);
+    }
+  }
+);
+export const fetchProductByUserID = createAsyncThunk<Product, string>(
+  'stepper/fetchProductByUserID',
+  async (userId, thunkAPI) => {
+    try {
+      const response = await axios.get(
+        `${apiBaseUrl}/create-product/user/${userId}`,
+        { withCredentials: true }
       );
+      return response.data as Product;
+    } catch (err: any) {
+      return thunkAPI.rejectWithValue(err.response?.data?.message || err.message);
     }
   }
 );
 
-// Update product step
 export const updateProductStep = createAsyncThunk<
   Product,
   { productId: string; stepNumber: number }
 >(
   'stepper/updateProductStep',
   async ({ productId, stepNumber }, thunkAPI) => {
     try {
-      const stepName = steps[stepNumber - 1]?.title || steps[0].title;
+      const stepName = stepsConfig[stepNumber - 1]?.title || stepsConfig[0].title;
```

Consider validating stepNumber before fallback to stepsConfig[0] to prevent silent fallback